src/main/scala/scalaz/netty/Client.scala

```diff
@@ -36,6 +36,7 @@ import _root_.io.netty.handler.codec._
 
 private[netty] final class Client(limit: Int) {
   // this isn't ugly or anything...
+  @volatile
```

Comment:
I read the code again, how about just move the channel as one of the constructor arguments for Server ? the Server factory method does not touch the server instance until the netty channel is gotten.